### PR TITLE
Fix #1044

### LINKED ISSUE #1044: In caching sha2 password xor, b length can be less than a.
https://github.com/sidorares/node-mysql2/blob/93df7a8f5c7f56d80285ebdd001508567cbe84e9/lib/auth_plugins/caching_sha2_password.js#L35

You could use xorRotating func to fix:
https://github.com/normano/node-mysql2/blob/72aaf4a07680d26edb5ac6bb828027207e97e004/lib/auth_41.js#L137

### PR DESCRIPTION
**Pull request by normano**
This will fix the password xor scramble length issue by using xorRotated (as specified by MySQL Engineering insider blog).

Issue: https://github.com/sidorares/node-mysql2/issues/1044

### CODE REVIEW

File: lib/auth_41.js [hunk 1]
```diff
@@ -57,17 +57,15 @@ function xor(a, b) {
...
   if (!password) {
     return Buffer.alloc(0);
   }
   const stage1 = sha1(password);
-  return exports.calculateTokenFromPasswordSha(
-    stage1,
-    scramble1,
-    scramble2
-  );
+  return exports.calculateTokenFromPasswordSha(stage1, scramble1, scramble2);
```
Comment (normano) [inline lib/auth_41.js]: I know I did not touch these, so I suppose there must be pre commit hooks to do some auto formatting.

Comment (sidorares) [reply lib/auth_41.js]: yes, there is prettier percommit hook

Comment (sidorares) [reply lib/auth_41.js]: via lint-staged

**Comment by sidorares** [comment]
LGTM, happy to merge once CI finish

**Pull request merged**